crypto-core/src/main/java/com/palantir/crypto/cipher/CipherStreamSupplier.java

```diff
@@ -30,12 +35,14 @@
     /**
      * Wraps the {@link InputStream} in a {@link CipherInputStream} which uses the given {@link Cipher} for decryption.
      */
-    CipherInputStream getInputStream(InputStream is, Cipher cipher);
+    CryptoInputStream getInputStream(String transformation, Properties props, InputStream is,
```

Comment:
I added those to be able to construct a `CryptoInputStream`, since those are things that the public constructor for that class wants. It does make this change significantly more invasive and messy though, so definitely open to changing my approach if there's a better way.